types/tx/msgs_test.go

```diff
@@ -0,0 +1,195 @@
+package tx_test
+
+import (
+	"testing"
+
+	"github.com/cosmos/gogoproto/proto"
+	"github.com/stretchr/testify/require"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/codec/types"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/tx"
+)
+
+func Test_SetMsg(t *testing.T) {
+	cases := []struct {
+		name   string
+		msg    sdk.Msg
+		expErr bool
+	}{
+		{
+			name:   "Set nil Msg",
+			msg:    nil,
+			expErr: true,
+		},
+		{
+			name:   "Set a valid message",
+			msg:    &DummyProtoMessage1{},
+			expErr: false,
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			actual, err := tx.SetMsg(tc.msg)
+			if tc.expErr {
+				require.Error(t, err)
+				return
+			}
+
+			expected := mustAny(tc.msg)
+			require.Equal(t, expected, actual)
+			require.Nil(t, err)
+		})
+	}
+}
```

Comment:
I find the map type easier to read and you can get rid of the name field. For example:
```go
	specs := map[string]struct{
        	msg    sdk.Msg
		expErr bool
	}{
	"Set nil Msg": {
		msg:    nil,
		expErr: true,
         },
	}
	for name, spec := range specs {
	t.Run(name, func(t *testing.T) {
	            
	})
	}
```

Reply:
Thanks for the comment. I really like how it looks with a map. I will start using map[string]struct{} instead struct{}.

Fixed